service/history/api/describeworkflow/api.go

```diff
@@ -178,13 +179,37 @@ func Invoke(
 				LastIndependentlyAssignedBuildId: ai.GetLastIndependentlyAssignedBuildId(),
 			}
 		}
-		if ai.CancelRequested {
-			p.State = enumspb.PENDING_ACTIVITY_STATE_CANCEL_REQUESTED
-		} else if ai.StartedEventId != common.EmptyEventID {
-			p.State = enumspb.PENDING_ACTIVITY_STATE_STARTED
+
+		p.State = workflow.ActivityState(ai)
+		p.Attempt = ai.Attempt
+		if p.Attempt < 1 {
+			p.Attempt = 1
+		}
+
+		if p.State == enumspb.PENDING_ACTIVITY_STATE_SCHEDULED && now.Before(ai.ScheduledTime.AsTime()) {
+			p.NextAttemptScheduleTime = ai.ScheduledTime
+		}
```

Comment:
Looks like you already have this check below.

Reply:
all this is gone, I will use refactored function from activity.go